### PR TITLE
Matic and Optimism fixes and integration tests added

### PR DESCRIPTION
**Pull request by YordanPavlov**
My previous PR have broken the Optimism and Matic exporters. In this PR - fixes along with integration tests.

### CODE REVIEW

File: blockchains/erc20/lib/timestamps_cache.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
     this.rangeSize = toBlock - fromBlock + 1;
+    this.web3Wrapper = web3Wrapper;
 
     const blockRequests = Array.from(
       { length: toBlock - fromBlock + 1 },
       (_, index) => ethClient.request(
         'eth_getBlockByNumber',
-        [fromBlock + index, false],
+        // Some Nodes would also accept decimal, but we convert to be on the safe side
+        [web3Wrapper.parseNumberToHex(fromBlock + index), false],
```
Comment (YordanPavlov) [inline blockchains/erc20/lib/timestamps_cache.js:14]: This works in decimal for Ethereum but needs to be in Hex for other chains.

File: blockchains/matic/lib/fetch_events.js [hunk 2]
```diff
@@ -32,11 +32,12 @@ const decodeFunctions = {
 
 };
 
-async function getPastEvents(web3Wrapper, fromBlock, toBlock) {
+async function getPastEvents(ethClient, web3Wrapper, fromBlock, toBlock) {
   const events = await getRawEvents(web3Wrapper, fromBlock, toBlock);
 
-  const decodedEvents = decodeEvents(web3Wrapper, events, new TimestampsCache(), decodeFunctions);
```
Comment (YordanPavlov) [inline blockchains/matic/lib/fetch_events.js:38]: I have previously missed to update the construction of TimestampsCache() in this context.

File: integration_test/src/matic.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const assert = require('assert');
+const worker = require('../../blockchains/matic/matic_worker');
+
+
+class MockExporter {
+  initPartitioner() {
+    // Dummy
+  }
+}
+describe('Matic worker test', function () {
```
Comment (YordanPavlov) [inline integration_test/src/matic.spec.js:10]: New Matic integration test.

File: integration_test/src/optimism-erc20.spec.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const assert = require('assert');
+const worker = require('../../blockchains/erc20/erc20_worker');
+
+
+class MockExporter {
+  initPartitioner() {
+    // Dummy
+  }
+}
+describe('Optimism worker test', function () {
```
Comment (YordanPavlov) [inline integration_test/src/optimism-erc20.spec.js:10]: New optimism integration test.

**Review by WonderBeat — APPROVED** [review]

**Review by spiderjako — APPROVED** [review]

**Pull request merged**